dataactvalidator/config/sqlrules/d38_detached_award_financial_assistance.sql

```diff
@@ -0,0 +1,8 @@
+-- FundingOfficeCode must be six characters long.
+
+SELECT
+    row_number,
+    funding_office_code
+FROM detached_award_financial_assistance
+WHERE submission_id = {0}
+    AND LENGTH(funding_office_code) != 6
```

Comment:
Since this is an optional field, NULL and '' should be valid entries (just check for `COALESCE(funding_office_code, '') != ''` or something)